internal/dataplane/parser/translate_httproute.go

```diff
@@ -305,40 +298,79 @@ func generateKongRouteFromHTTPRouteMatches(
 	// stripPath needs to be disabled by default to be conformant with the Gateway API
 	r.StripPath = kong.Bool(false)
 
-	// attach the plugins to be applied to the given route
-	if len(plugins) != 0 {
-		if r.Plugins == nil {
-			r.Plugins = make([]kong.Plugin, 0, len(plugins))
+	seenMethods := make(map[string]struct{})
+
+	var hasRedirectFilter bool
+	for _, f := range filters {
```

Comment:
Take it or leave it: for this types of "for loop in order to find an element" you can use [`lo.Find()`](https://github.com/samber/lo#find)

Reply:
I need to use the more `lo` library, it is a cornucopia of wonders 😀